### ISSUE TITLE
[BUG]: ´IntegerVar´ doesn't correct floating point numbers

### ISSUE THREAD

**Issue by Kroppeb** [thieu1995/mealpy#133]
### Description of the bug

I've noticed that problem parameters that are an ´IntegerVar´ will in the first round of ´OriginalGSKA´ for example use integers, but subsequent rounds will use various floating point numbers.

### Steps To Reproduce

´´´python
import typing
import mealpy

def main():
    print(mealpy.__version__)  # 3.0.1

    # This function is minimal at x = 2/3 with a loss of 0, but the smallest loss for any integer is at x = 1 with a loss of 1
    problem = {
        "obj_func": lambda x: (3 * x - 2) ** 2,
        "bounds": mealpy.IntegerVar(
            lb=-100,
            ub=100,
        ),
        "minmax": "min",
        "log_to": None,
    }
    model = mealpy.human_based.GSKA.OriginalGSKA(epoch=30, pop_size=25)
    best = model.solve(problem)
    best_position = best.solution
    best_loss = typing.cast(float, best.target.fitness)

    # Expected output:
    # Best solution: [1.0], Loss: 1.0
    print(f"Best solution: {best_position}, Loss: {best_loss}")
    # Actual output (example):
    # Best solution: [0.66667098], Loss: 1.67278794715044e-10

if __name__ == "__main__":
    main()

´´´

**Comment by thieu1995** [comment]
Hi @Kroppeb,

I have made a note in https://github.com/thieu1995/mealpy?tab=readme-ov-file#simple-benchmark-function
Please use class definition for your problem if your decision variable is not float.
